src/kvp.rs

```diff
@@ -205,6 +206,159 @@ impl EmitKVPLayer {
         let encoded_kvp_flattened: Vec<u8> = encoded_kvp.concat();
         self.send_event(encoded_kvp_flattened);
     }
+
+    /// Processes and sends a health report event based on the provided `health_report` field.
+    ///
+    /// This function extracts the `reason` field (if present) and constructs a properly
+    /// formatted health report key-value pair (KVP) using either
+    /// 'build_success_health_report' or 'build_failure_health_report', depending on
+    /// the value of `health_report`. The resulting KVP is then encoded using
+    /// 'encode_kvp_item' and sent via `send_event()`.
+    ///
+    /// This function is called exclusively from [`on_event`] when a `health_report`
+    /// flag is detected, ensuring that health-related events are processed separately
+    /// from standard event logging.
+    ///
+    /// # Parameters
+    /// - `event`: The original tracing event containing metadata.
+    /// - `health_report`: The health report status as a string (e.g., `"Success"` or `"Failure"`).
+    /// - `timestamp`: The event timestamp as a formatted string.
+    ///
+    /// # Supported Tracing Fields
+    ///
+    /// - `reason`: A human-readable failure message. Required for `"failure"` reports.
+    /// - `optional_key_value`: An optional single key=value pair to include in `"success"` reports.
+    /// - `extra`: Optional field containing additional key-value pairs for `"failure"` reports.
+    ///
+    /// # `optional_key_value` Format
+    ///
+    /// A single key=value string to be flattened into the final KVP report:
+    ///
+    /// ```rust
+    /// tracing::info!(
+    ///     health_report = "success",
+    ///     optional_key_value = "origin=imds",
+    ///     "Provisioning succeeded"
+    /// );
+    /// ```
+    ///
+    /// # `extra` Field Format
+    ///
+    /// The `extra` field can be provided in one of two formats:
+    /// 1. **Semicolon-delimited string** (easiest to write):
+    ///
+    /// ```rust
+    /// tracing::info!(
+    ///     health_report = "failure",
+    ///     reason = "SSH key write failed",
+    ///     extra = "step=write_authorized_keys;component=ssh",
+    ///     "Provisioning failed"
+    /// );
+    /// ```
+    ///
+    /// 2. **JSON-formatted string** (machine-friendly or more structured):
+    ///
+    /// ```rust
+    /// tracing::info!(
+    ///     health_report = "failure",
+    ///     reason = "Network timeout",
+    ///     extra = r#"{"step":"imds_query","region":"westus"}"#,
+    ///     "Provisioning failed"
+    /// );
+    /// ```
+    ///
+    /// Both formats will be parsed into key-value pairs and included in the final KVP report.
+    fn handle_health_report(
+        &self,
```

Comment:
this should just be taking the encoded string that is output from ReportableError